src/Components/Server/src/ComponentHub.cs

```diff
@@ -220,7 +221,18 @@ public async ValueTask ReceiveByteArray(int id, byte[] data)
                 return;
             }
 
-            await circuitHost.ReceiveByteArray(id, data);
+            _ = circuitHost.ReceiveByteArray(id, data);
+        }
+
+        public async ValueTask<bool> ReceiveJSDataChunk(long streamId, long chunkId, byte[] chunk, string error)
+        {
+            var circuitHost = await GetActiveCircuitAsync();
+            if (circuitHost == null)
+            {
+                return false;
+            }
+
+            return await circuitHost.ReceiveJSDataChunk(streamId, chunkId, chunk, error);
```

Comment:
Since reviews are wrapping up, just calling this out from an offline conversation, this await will block the entire Hub until the operation completes which affects user interactivity

Reply:
Just to clarify/confirm, "this await will block the entire Hub" is referring to blocking that specific circuit, or _all_ circuits on the server?

Reply:
Sorry, just that connection

Reply:
Thanks for the clarification. In that case, we're acknowledging the blocking here, but are electing to keep it. This is because we're going to be blocking the circuit regardless when we call into `circuitHost.ReceiveDataChunk` [which calls for `Renderer.Dispatcher.InvokeAsync` ensuring we're on the main circuit thread](https://github.com/dotnet/aspnetcore/pull/33491/commits/9e63bb6e06d00425a0c23bc83dc4db266d959661#diff-c08c874d10203eb22ffb6d6417cb0fcc90d4984f9486ae7c585024b458e06b5fR431-R434). This is done to ensure the server/client runs in the same synchronization context (as is done with all the other endpoints). Additionally the `ReceiveJSDataChunk` is a quick process, it should return near instantly in most cases. Lastly, we're utilizing the return value as a heartbeat for the transfer process, and without it would likely need to setup a separate endpoint to handle that functionality.